Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,291 @@
+---
+title: Implicitly-scoped qubit allocation
+description: Provides more convenient syntax for allocating qubits.
+author: Sarah Marshall
+date: TBD
+---
+
+# Proposal
+
+1. Rename `using` and `borrowing` keywords to `use` and `borrow`.
+2. Allow `use` and `borrow` statements to be followed by either a block `{ ... }` or a semicolon `;`.
```

Comment:
The more I think about this, the more I would like to go back to the original proposal with the scope statement and without the block `use` statement. One of the primary goals of this proposal is to avoid forcing users to decide between excessive block nesting and unnecessary tuple destructuring. But by removing the scope statement, and having a block `use` statement instead, we continue to have that exact problem whenever you want to allocate more than one qubit variable in a short-lived scope:

```qsharp
X(q1);
use (q2, q3) = (Qubit(), Qubit()) {
    // ...
}
return M(q1);
```

I would prefer instead:

```qsharp
X(q1);
{
    use q2 = Qubit();
    use q3 = Qubit();
    // ...
}
return M(q1);
```

@bettinaheim @cgranade @viktorveis

Reply:
I like the scope syntax more than the `use` blocks as I think it is more syntactically consistent with what we already have the in language. The `use q = Qubit();` is consistent with how we define other symbols. And the scope syntax could almost be seen as a generalization of how we declare scopes for other language constructs like callables and for loops. This syntax could be built upon to possible define lambdas, if we choose to go down that route.

Reply:
After discussing this with @bettinaheim offline, we agreed that it was better to avoid making major changes to the existing block statement syntax until we know more about how the non-block syntax will be used in practice, and if there is actually a need to better support more advanced allocation with an explicit scope.